oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -211,11 +211,31 @@ static FullGCMode getFullGcMode() {
         return fullGcMode;
     }
 
+    /**
+     * Set the full GC mode to be used according to the provided configuration value.
+     * The configuration value will be ignored if it is set to any other values than the supported ones.
+     * @param fullGcModeConfig
+     */
+    static void setFullGcMode(int fullGcModeConfig) {
```

Comment:
We decided to put this logic inside the `FullGCMode` enum.
`FullGCMode` enum should have a constructor with both `int` & `string` value and int value should match with the OSGI config.

Reply:
I think this is outside the scope of this ticket.
Doing this change now would also require refactoring the unit tests that use fullGCMode. 
I suggest that we do that in a separate refactoring ticket, since there are a lot of tests involved and we may also consider other changes to the unit tests when we do the refactoring.